src/decode.rs

```diff
@@ -797,14 +798,28 @@ unsafe fn read_pal_indices(
             pal_tmp[offset..][..len].fill(filler);
         }
     }
-    (pal_dsp.pal_idx_finish)(
-        pal_idx.unwrap_or(pal_tmp).as_mut_ptr(),
-        pal_tmp.as_ptr(),
-        bw4 as c_int * 4,
-        bh4 as c_int * 4,
-        w4 as c_int * 4,
-        h4 as c_int * 4,
-    );
+
+    if let Some(pal_idx) = &pal_idx {
+        let read_len = bw4 * 2 * bh4 * 4;
+        debug_assert!(pal_idx.len() >= read_len);
+    }
+    let read_len = bw4 * 4 * bh4 * 4;
+    debug_assert!(pal_tmp.len() >= read_len);
+
+    // SAFETY: Unsafe asm call. `pal_idx` is at least `(bw * 2) * (bh * 4)`
```

Comment:
```suggestion
    // SAFETY: Unsafe asm call. `pal_idx` is at least `(bw4 * 2) * (bh4 * 4)`
```

That's what `read_len` above is; one of these is wrong, right?

Reply:
Ah yeah that should be `bw4` and `bh4`, will fix.